File: app/src/main/java/com/mmjang/ankihelper/ui/ai/AIDictionaryConfigListActivity.java
package com.mmjang.ankihelper.ui.ai;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NavUtils;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import com.mmjang.ankihelper.R;
import com.mmjang.ankihelper.data.ai.AIConfigRepository;
import com.mmjang.ankihelper.data.ai.AIDictionaryConfig;

import java.util.List;

public class AIDictionaryConfigListActivity extends AppCompatActivity implements AIDictionaryConfigAdapter.OnItemClickListener {
    private RecyclerView recyclerView;
    private AIDictionaryConfigAdapter adapter;
    private List<AIDictionaryConfig> dictionaryConfigList;
    private FloatingActionButton fabAddDictionary;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        if (com.mmjang.ankihelper.data.Settings.getInstance(this).getPinkThemeQ()) {
            setTheme(R.style.AppThemePink);
        }
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ai_dictionary_config_list);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        initViews();
        loadDictionaryConfigs();
    }

    @Override
    protected void onResume() {
        super.onResume();
        loadDictionaryConfigs(); // Refresh the list when returning from editor
    }

    private void initViews() {
        recyclerView = findViewById(R.id.recycler_view_dictionary_configs);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        fabAddDictionary = findViewById(R.id.fab_add_dictionary);
        fabAddDictionary.setOnClickListener(v -> {
            Intent intent = new Intent(AIDictionaryConfigListActivity.this, AIDictionaryConfigEditorActivity.class);
            startActivity(intent);
        });
    }

    private void loadDictionaryConfigs() {
        dictionaryConfigList = AIConfigRepository.getAllAIDictionaryConfigs();
        adapter = new AIDictionaryConfigAdapter(dictionaryConfigList, this);
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onItemClick(AIDictionaryConfig config) {
        Intent intent = new Intent(AIDictionaryConfigListActivity.this, AIDictionaryConfigEditorActivity.class);
        intent.putExtra("dictionary_config_id", config.getId());
        startActivity(intent);
    }

    @Override
    public void onDeleteClick(AIDictionaryConfig config) {
        // Show confirmation dialog
        new AlertDialog.Builder(this)
                .setTitle("Delete Configuration")
                .setMessage("Are you sure you want to delete \"" + config.getDictionaryName() + "\"?")
                .setPositiveButton("Delete", (dialog, which) -> {
                    // Delete the configuration
                    AIConfigRepository.deleteAIDictionaryConfig(config.getId());
                    // Refresh the list
                    loadDictionaryConfigs();
                })
                .setNegativeButton("Cancel", null)
                .show();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/lmyby/ankihelper/ui/ai/AIDictionaryConfigListActivity.kt
package com.lmyby.ankihelper.ui.ai

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.NavUtils
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.lmyby.ankihelper.R
import com.lmyby.ankihelper.data.Settings
import com.lmyby.ankihelper.data.ai.AIConfigRepository
import com.lmyby.ankihelper.data.ai.AIDictionaryConfig

/**
 * Activity for managing AI dictionary configuration list
 * Converted to Kotlin as part of Phase 5 activity migration
 */
class AIDictionaryConfigListActivity : AppCompatActivity(), AIDictionaryConfigAdapter.OnItemClickListener {

    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: AIDictionaryConfigAdapter
    private lateinit var dictionaryConfigList: List<AIDictionaryConfig>
    private lateinit var fabAddDictionary: FloatingActionButton

    override fun onCreate(savedInstanceState: Bundle?) {
        if (Settings.getInstance(this).getPinkThemeQ()) {
            setTheme(R.style.AppThemePink)
        }
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ai_dictionary_config_list)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        initViews()
        loadDictionaryConfigs()
    }

    override fun onResume() {
        super.onResume()
        loadDictionaryConfigs() // Refresh the list when returning from editor
    }

    private fun initViews() {
        recyclerView = findViewById(R.id.recycler_view_dictionary_configs)
        recyclerView.layoutManager = LinearLayoutManager(this)

        fabAddDictionary = findViewById(R.id.fab_add_dictionary)
        fabAddDictionary.setOnClickListener {
            val intent = Intent(this@AIDictionaryConfigListActivity, AIDictionaryConfigEditorActivity::class.java)
            startActivity(intent)
        }
    }

    private fun loadDictionaryConfigs() {
        dictionaryConfigList = AIConfigRepository.getAllAIDictionaryConfigs()
        adapter = AIDictionaryConfigAdapter(dictionaryConfigList, this)
        recyclerView.adapter = adapter
    }

    override fun onItemClick(config: AIDictionaryConfig) {
        val intent = Intent(this@AIDictionaryConfigListActivity, AIDictionaryConfigEditorActivity::class.java)
        intent.putExtra("dictionary_config_id", config.id)
        startActivity(intent)
    }

    override fun onDeleteClick(config: AIDictionaryConfig) {
        // Show confirmation dialog
        AlertDialog.Builder(this)
            .setTitle("Delete Configuration")
            .setMessage("Are you sure you want to delete \"${config.dictionaryName}\"?")
            .setPositiveButton("Delete") { _, _ ->
                // Delete the configuration
                AIConfigRepository.deleteAIDictionaryConfig(config.id)
                // Refresh the list
                loadDictionaryConfigs()
            }
            .setNegativeButton("Cancel", null)
            .show()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this)
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
